apps/console/src/features/console-settings/components/console-roles/console-roles-edit/console-role-permissions.tsx

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    AlertInterface,
+    AlertLevels,
+    IdentifiableComponentInterface,
+    LoadableComponentInterface,
+    RolePermissionInterface,
+    RolesInterface
+} from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import {
+    EmphasizedSegment,
+    Heading,
+    PrimaryButton
+} from "@wso2is/react-components";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { updateRoleDetails } from "../../../../roles/api/roles";
+import { Schemas } from "../../../../roles/constants/role-constants";
+import { CreateRolePermissionInterface, PatchRoleDataInterface } from "../../../../roles/models/roles";
+import useGetAPIResourceCollections from "../../../api/use-get-api-resource-collections";
+import {
+    APIResourceCollectionInterface,
+    APIResourceCollectionPermissionCategoryInterface,
+    APIResourceCollectionPermissionScopeInterface
+} from "../../../models/console-roles";
+import { SelectedPermissionCategoryInterface, SelectedPermissionsInterface } from "../../../models/permissions-ui";
+import transformResourceCollectionToPermissions from "../../../utils/transform-resource-collection-to-permissions";
+import CreateConsoleRoleWizardPermissionsForm from
+    "../create-console-role-wizard/create-console-role-wizard-permissions-form";
+
+/**
+ * Props interface of {@link ConsoleRolePermissions}
```

Comment:
is @link a valid TSDoc tag?

Reply:
Yes. It is like a valid tag.

https://tsdoc.org/pages/tags/link/